File: problems/0001_two_sum/v2_hashmap_one_pass/solution.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nmap = {}
        n = len(nums)

        for i in range(n):
            nmap[nums[i]] = i
        
        for i in range(n):
            complement = target - nums[i]
            if complement in nmap and nmap[complement] != i:
                return [i, nmap[complement]]
        
        return []
    
# Example usage:
nums = [3,2,4]
target = 6
solution = Solution()
result = solution.twoSum(nums, target)
print(result)  # Output: [1, 2]

File: problems/0001_two_sum/v2_hashmap_one_pass/solution.ts
namespace HashMap {
    function twoSum(nums: number[], target: number): number[] {
        const nmap = new Map<number, number>();
        const n = nums.length;

        for (let i = 0; i < n; i++) {
            nmap.set(nums[i], i);
        }

        for (let i = 0; i < n; i++) {
            const complement = target - nums[i];
            if (nmap.has(complement) && nmap.get(complement)! !== i) {
                return [i, nmap.get(complement)!];
            }
        }

        return [];
    }

    // Example usage:
    const testNums = [3, 2, 4];
    const testTarget = 6;
    const result = twoSum(testNums, testTarget);
    console.log(result); // Output: [1, 2]
}